modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -627,37 +516,57 @@ private void dropTableLocally(String name, UUID tblId, List<List<ClusterNode>> a
     @Override public CompletableFuture<Void> alterTableAsync(String name, Consumer<TableChange> tableChange) {
         CompletableFuture<Void> tblFut = new CompletableFuture<>();
 
-        listen(TableEvent.ALTER, new EventListener<>() {
-            @Override public boolean notify(@NotNull TableEventParameters parameters, @Nullable Throwable e) {
-                String tableName = parameters.tableName();
-
-                if (!name.equals(tableName))
-                    return false;
-
-                if (e == null)
-                    tblFut.complete(null);
-                else
-                    tblFut.completeExceptionally(e);
-
-                return true;
+        tableAsync(name, true).thenAccept(tbl -> {
+            if (tbl == null) {
+                tblFut.completeExceptionally(new IgniteException(
+                    LoggerMessageHelper.format("Table already exists [name={}]", name)));
```

Comment:
It is a wrong message.
Table does not exist and cannot be altered.

Reply:
Fixed.